### ISSUE TITLE
Panic in Constellation - unable to find pipeline

### ISSUE THREAD

**Issue by connorimes** [servo/servo#9202]
Not entirely sure how this occurred, I just ran a release build with the 'energy-profiling' feature and loading google.com.  I wasn't really paying attention, but I think it crashed while I was closing servo.

<pre>
./mach run -r http://www.google.com
Started energy monitoring from: Intel RAPL
thread 'Constellation' panicked at 'unable to find pipeline - this is a bug', ../src/libcore/option.rs:335
stack backtrace:
ERROR:js::rust: Error at https://apis.google.com/_/scs/abc-static/_/js/k=gapi.gapi.en.0qt5d8yawpg.O/m=gapi_iframes,googleapis_client,plusone/rt=j/sv=1/d=1/ed=1/am=AAQ/rs=AItRSTPlCaUSPAbLD881PE1fL-khxRsvgg/cb=gapi.loaded_0:202:157: qa.Xb is not a function

   1:     0x56111fac2e50 - sys::backtrace::tracing::imp::write::h92808652e552bec0MZt
   2:     0x56111fac5cc5 - panicking::log_panic::_<closure>::closure.41184
   3:     0x56111fac5740 - panicking::log_panic::hc4038fe8a2a609d626x
   4:     0x56111faaec13 - sys_common::unwind::begin_unwind_inner::hd5a3cc6e3320d6b8lSs
   5:     0x56111faaf2c8 - sys_common::unwind::begin_unwind_fmt::h7eec50b09af6e91crRs
   6:     0x56111fac24b1 - rust_begin_unwind
   7:     0x56111faf3bff - panicking::panic_fmt::h871c0ef100fda9e9LFK
   8:     0x56111e297bbc - constellation::_<impl>::handle_request::handle_request::h16596261501807853628
   9:     0x56111e2838c7 - export::compositing::constellation::_<impl>::start::_<closure>::closure.23378
  10:     0x56111e283123 - sys_common::unwind::try::try_fn::try_fn::h3215051430881614784
  11:     0x56111fac22f8 - __rust_try
  12:     0x56111fabe6bb - sys_common::unwind::try::inner_try::h3990fffe7b775df5TOs
  13:     0x56111e28510a - boxed::_<impl>::call_box::call_box::h10952870076713712831
  14:     0x56111fac4623 - sys::thread::_<impl>::new::thread_start::h05427eb97025f9aae9w
  15:     0x7ffab6af66a9 - start_thread
  16:     0x7ffab85f9eec - clone
  17:                0x0 - <unknown>
thread 'LayoutTask PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(1) }' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError', ../src/libcore/result.rs:741
stack backtrace:
   1:     0x56111fac2e50 - sys::backtrace::tracing::imp::write::h92808652e552bec0MZt
   2:     0x56111fac5cc5 - panicking::log_panic::_<closure>::closure.41184
   3:     0x56111fac5740 - panicking::log_panic::hc4038fe8a2a609d626x
   4:     0x56111faaec13 - sys_common::unwind::begin_unwind_inner::hd5a3cc6e3320d6b8lSs
   5:     0x56111faaf2c8 - sys_common::unwind::begin_unwind_fmt::h7eec50b09af6e91crRs
   6:     0x56111fac24b1 - rust_begin_unwind
   7:     0x56111faf3bff - panicking::panic_fmt::h871c0ef100fda9e9LFK
   8:     0x56111e3f2566 - layout_task::_<impl>::handle_request::hfdd05d31725a44280pq
   9:     0x56111e3bd49a - sys_common::unwind::try::try_fn::try_fn::h13501037888748617246
  10:     0x56111fac22f8 - __rust_try
  11:     0x56111fabe6bb - sys_common::unwind::try::inner_try::h3990fffe7b775df5TOs
  12:     0x56111e3be47a - boxed::_<impl>::call_box::call_box::h4168095935381687278
  13:     0x56111fac4623 - sys::thread::_<impl>::new::thread_start::h05427eb97025f9aae9w
  14:     0x7ffab6af66a9 - start_thread
  15:     0x7ffab85f9eec - clone
  16:                0x0 - <unknown>
thread 'PaintTask PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(1) }thread '' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError', ../src/libcore/result.rs:741
stack backtrace:
  thread 'LayoutWorker worker 3/3 1:    thread 'LayoutWorker worker 2/3 0x56111fac2e50' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError - sys' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError::backtraceLayoutWorker worker 1/3::tracing', ../src/libcore/result.rs::imp:741::write' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError::h92808652e552bec0MZt', ../src/libcore/result.rs

  :741
 2', ../src/libcore/result.rs:    :741 0x56111fac5cc5 - panicking::log_panic::_<closure>::closure.41184
   3:     0x56111fac5740 - panicking::log_panic::hc4038fe8a2a609d626x

   4:     0x56111faaec13 - sys_common::unwind::begin_unwind_inner::hd5a3cc6e3320d6b8lSs
   5:     0x56111faaf2c8 - sys_common::unwind::begin_unwind_fmt::h7eec50b09af6e91crRs
   6:     0x56111fac24b1 - rust_begin_unwind
   7:     0x56111faf3bff - panicking::panic_fmt::h871c0ef100fda9e9LFK
   8:     0x56111ec628f9 - paint_task::_<impl>::start::start::h1382422297082552915
   9:     0x56111ec54511 - sys_common::unwind::try::try_fn::try_fn::h5202191907091808115
  10:     0x56111fac22f8 - __rust_try
  11:     0x56111fabe6bb - sys_common::unwind::try::inner_try::h3990fffe7b775df5TOs
  12:     0x56111ec5505a - boxed::_<impl>::call_box::call_box::h15997509081803256275
  13:     0x56111fac4623 - sys::thread::_<impl>::new::thread_start::h05427eb97025f9aae9w
  14:     0x7ffab6af66a9 - start_thread
  15:     0x7ffab85f9eec - clone
  16:                0x0 - <unknown>
stack backtrace:
   1:   thread 'thread 'thread 'PaintWorker' panicked at 'called `Result::unwrap()` on an `Err` value: RecvErrorPaintWorkerPaintWorker', ' panicked at '' panicked at '../src/libcore/result.rscalled `Result::unwrap()` on an `Err` value: RecvErrorcalled `Result::unwrap()` on an `Err` value: RecvError:', ', ../src/libcore/result.rs../src/libcore/result.rs::741741

741
  0x56111fac2e50 - sys::backtrace::tracing::imp::write::h92808652e552bec0MZt
   2:     0x56111fac5cc5 - panicking::log_panic::_<closure>::closure.41184
   3:     0x56111fac5740 - panicking::log_panic::hc4038fe8a2a609d626x
   4:     0x56111faaec13 - sys_common::unwind::begin_unwind_inner::hd5a3cc6e3320d6b8lSs
   5:     0x56111faaf2c8 - sys_common::unwind::begin_unwind_fmt::h7eec50b09af6e91crRs
   6:     0x56111fac24b1 - rust_begin_unwind
   7:     0x56111faf3bff - panicking::panic_fmt::h871c0ef100fda9e9LFK
   8:     0x56111e3e267b - sys_common::unwind::try::try_fn::try_fn::h3610648465701720216
   9:     0x56111fac22f8 - __rust_try
  10:     0x56111fabe6bb - sys_common::unwind::try::inner_try::h3990fffe7b775df5TOs
  11:     0x56111e3e36f9 - boxed::_<impl>::call_box::call_box::h3584596844243775600
  12:     0x56111fac4623 - sys::thread::_<impl>::new::thread_start::h05427eb97025f9aae9w
  13:     0x7ffab6af66a9 - start_thread
  14:     0x7ffab85f9eec - clone
  15:                0x0 - <unknown>
stack backtrace:
   1:     0x56111fac2e50 - sys::backtrace::tracing::imp::write::h92808652e552bec0MZt
   2:     0x56111fac5cc5 - panicking::log_panic::_<closure>::closure.41184
   3:     0x56111fac5740 - panicking::log_panic::hc4038fe8a2a609d626x
   4:     0x56111faaec13 - sys_common::unwind::begin_unwind_inner::hd5a3cc6e3320d6b8lSs
   5:     0x56111faaf2c8 - sys_common::unwind::begin_unwind_fmt::h7eec50b09af6e91crRs
   6:     0x56111fac24b1 - rust_begin_unwind
   7:     0x56111faf3bff - panicking::panic_fmt::h871c0ef100fda9e9LFK
   8:     0x56111e3e267b - sys_common::unwind::try::try_fn::try_fn::h3610648465701720216
   9:     0x56111fac22f8 - __rust_try
  10:     0x56111fabe6bb - sys_common::unwind::try::inner_try::h3990fffe7b775df5TOs
  11:     0x56111e3e36f9 - boxed::_<impl>::call_box::call_box::h3584596844243775600
  12:     0x56111fac4623 - sys::thread::_<impl>::new::thread_start::h05427eb97025f9aae9w
  13:     0x7ffab6af66a9 - start_thread
  14:     0x7ffab85f9eec - clone
  15:                0x0 - <unknown>
stack backtrace:
   1:     0x56111fac2e50 - sys::backtrace::tracing::imp::write::h92808652e552bec0MZt
   2:     0x56111fac5cc5 - panicking::log_panic::_<closure>::closure.41184
   3:     0x56111fac5740 - panicking::log_panic::hc4038fe8a2a609d626x
   4:     0x56111faaec13 - sys_common::unwind::begin_unwind_inner::hd5a3cc6e3320d6b8lSs
   5:     0x56111faaf2c8 - sys_common::unwind::begin_unwind_fmt::h7eec50b09af6e91crRs
   6:     0x56111fac24b1 - rust_begin_unwind
   7:     0x56111faf3bff - panicking::panic_fmt::h871c0ef100fda9e9LFK
   8:     0x56111e3e267b - sys_common::unwind::try::try_fn::try_fn::h3610648465701720216
   9:     0x56111fac22f8 - __rust_try
  10:     0x56111fabe6bb - sys_common::unwind::try::inner_try::h3990fffe7b775df5TOs
  11:     0x56111e3e36f9 - boxed::_<impl>::call_box::call_box::h3584596844243775600
  12:     0x56111fac4623 - sys::thread::_<impl>::new::thread_start::h05427eb97025f9aae9w
  13:     0x7ffab6af66a9 - start_thread
  14:     0x7ffab85f9eec - clone
  15:                0x0 - <unknown>
stack backtrace:
   1:     0x56111fac2e50 - sys::backtrace::tracing::imp::write::h92808652e552bec0MZt
   2:     0x56111fac5cc5 - panicking::log_panic::_<closure>::closure.41184
   3:     0x56111fac5740 - panicking::log_panic::hc4038fe8a2a609d626x
   4:     0x56111faaec13 - sys_common::unwind::begin_unwind_inner::hd5a3cc6e3320d6b8lSs
   5:     0x56111faaf2c8 - sys_common::unwind::begin_unwind_fmt::h7eec50b09af6e91crRs
   6:     0x56111fac24b1 - rust_begin_unwind
   7:     0x56111faf3bff - panicking::panic_fmt::h871c0ef100fda9e9LFK
   8:     0x56111ed2df0c - paint_task::_<impl>::main::h82ba22623cf4f1d15un
   9:     0x56111ed2b7b6 - sys_common::unwind::try::try_fn::try_fn::h12757664780853574016
  10:     0x56111fac22f8 - __rust_try
  11:     0x56111fabe6bb - sys_common::unwind::try::inner_try::h3990fffe7b775df5TOs
  12:     0x56111ed2c056 - boxed::_<impl>::call_box::call_box::h3514589817228806917
  13:     0x56111fac4623 - sys::thread::_<impl>::new::thread_start::h05427eb97025f9aae9w
  14:     0x7ffab6af66a9 - start_thread
  15:     0x7ffab85f9eec - clone
  16:                0x0 - <unknown>
stack backtrace:
   1:     0x56111fac2e50 - sys::backtrace::tracing::imp::write::h92808652e552bec0MZt
   2:     0x56111fac5cc5 - panicking::log_panic::_<closure>::closure.41184
   3:     0x56111fac5740 - panicking::log_panic::hc4038fe8a2a609d626x
   4:     0x56111faaec13 - sys_common::unwind::begin_unwind_inner::hd5a3cc6e3320d6b8lSs
   5:     0x56111faaf2c8 - sys_common::unwind::begin_unwind_fmt::h7eec50b09af6e91crRs
   6:     0x56111fac24b1 - rust_begin_unwind
   7:     0x56111faf3bff - panicking::panic_fmt::h871c0ef100fda9e9LFK
   8:     0x56111ed2df0c - paint_task::_<impl>::main::h82ba22623cf4f1d15un
   9:     0x56111ed2b7b6 - sys_common::unwind::try::try_fn::try_fn::h12757664780853574016
  10:     0x56111fac22f8 - __rust_try
  11:     0x56111fabe6bb - sys_common::unwind::try::inner_try::h3990fffe7b775df5TOs
  12:     0x56111ed2c056 - boxed::_<impl>::call_box::call_box::h3514589817228806917
  13:     0x56111fac4623 - sys::thread::_<impl>::new::thread_start::h05427eb97025f9aae9w
  14:     0x7ffab6af66a9 - start_thread
  15:     0x7ffab85f9eec - clone
  16:                0x0 - <unknown>
stack backtrace:
   1:     0x56111fac2e50 - sys::backtrace::tracing::imp::write::h92808652e552bec0MZt
   2:     0x56111fac5cc5 - panicking::log_panic::_<closure>::closure.41184
   3:     0x56111fac5740 - panicking::log_panic::hc4038fe8a2a609d626x
   4:     0x56111faaec13 - sys_common::unwind::begin_unwind_inner::hd5a3cc6e3320d6b8lSs
   5:     0x56111faaf2c8 - sys_common::unwind::begin_unwind_fmt::h7eec50b09af6e91crRs
   6:     0x56111fac24b1 - rust_begin_unwind
   7:     0x56111faf3bff - panicking::panic_fmt::h871c0ef100fda9e9LFK
   8:     0x56111ed2df0c - paint_task::_<impl>::main::h82ba22623cf4f1d15un
   9:     0x56111ed2b7b6 - sys_common::unwind::try::try_fn::try_fn::h12757664780853574016
  10:     0x56111fac22f8 - __rust_try
  11:     0x56111fabe6bb - sys_common::unwind::try::inner_try::h3990fffe7b775df5TOs
  12:     0x56111ed2c056 - boxed::_<impl>::call_box::call_box::h3514589817228806917
  13:     0x56111fac4623 - sys::thread::_<impl>::new::thread_start::h05427eb97025f9aae9w
  14:     0x7ffab6af66a9 - start_thread
  15:     0x7ffab85f9eec - clone
  16:                0x0 - <unknown>
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: "SendError(..)"', ../src/libcore/result.rs:741
stack backtrace:
   1:     0x56111fac2e50 - sys::backtrace::tracing::imp::write::h92808652e552bec0MZt
   2:     0x56111fac5cc5 - panicking::log_panic::_<closure>::closure.41184
   3:     0x56111fac5740 - panicking::log_panic::hc4038fe8a2a609d626x
   4:     0x56111faaec13 - sys_common::unwind::begin_unwind_inner::hd5a3cc6e3320d6b8lSs
   5:     0x56111faaf2c8 - sys_common::unwind::begin_unwind_fmt::h7eec50b09af6e91crRs
   6:     0x56111fac24b1 - rust_begin_unwind
   7:     0x56111faf3bff - panicking::panic_fmt::h871c0ef100fda9e9LFK
   8:     0x56111e22281c - compositor::_<impl>::handle_browser_message::handle_browser_message::h4735069503330065454
   9:     0x56111e21bdf4 - compositor::_<impl>::handle_events::handle_events::h12991668205173167111
  10:     0x56111e253bbc - _<impl>::handle_events::h99619fd1cfed3bddTda
  11:     0x56111e2080e5 - main::h309fd0ef92de1e9cOaa
  12:     0x56111fac54e4 - sys_common::unwind::try::try_fn::h7862647019611204684
  13:     0x56111fac22f8 - __rust_try
  14:     0x56111fac5186 - rt::lang_start::h48fbdbf4c3b8b04ca4x
  15:     0x7ffab8513a3f - __libc_start_main
  16:     0x56111e1fbc78 - _start
  17:                0x0 - <unknown>
thread '<main>' panicked at 'You should have disposed of the pixmap properly with destroy()! This pixmap will leak!', /local/servo/.cargo/git/checkouts/rust-layers-c02709c0157e02cd/master/src/platform/linux/surface.rs:166
stack backtrace:
   1:     0x56111fac2e50 - sys::backtrace::tracing::imp::write::h92808652e552bec0MZt
   2:  thread '   0x56111fac5cc5 - LayoutTask PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(0) }panicking::' panicked at 'called `Result::unwrap()` on an `Err` value: RecvErrorlog_panic::', ../src/libcore/result.rs_<:741closure>
::closure.41184
   3:     0x56111fac5740 - panicking::log_panic::hc4038fe8a2a609d626x
   4:     0x56111faaeb38 - sys_common::unwind::begin_unwind_inner::hd5a3cc6e3320d6b8lSs
   5:     0x56111f77a75f - sys_common::unwind::begin_unwind::begin_unwind::h11257214602500465916
   6:     0x56111f77a860 - platform::linux::surface::_<impl>::drop::hf82c15b6396387cfXac
   7:     0x56111e2184b7 - alloc..rc..Rc<layers..layers..Layer<compositing..compositor_layer..CompositorData>>::drop.29345::h5614f0c88e793b54
   8:     0x56111e218605 - collections..vec..Vec<alloc..rc..Rc<layers..layers..Layer<compositing..compositor_layer..CompositorData>>>::drop.29357::h084f2db6a61851c5
   9:     0x56111e218394 - alloc..rc..Rc<layers..layers..Layer<compositing..compositor_layer..CompositorData>>::drop.29345::h5614f0c88e793b54
  10:     0x56111e218605 - collections..vec..Vec<alloc..rc..Rc<layers..layers..Layer<compositing..compositor_layer..CompositorData>>>::drop.29357::h084f2db6a61851c5
  11:     0x56111e218394 - alloc..rc..Rc<layers..layers..Layer<compositing..compositor_layer..CompositorData>>::drop.29345::h5614f0c88e793b54
  12:     0x56111e21a4d1 - compositing..compositor..IOCompositor<app..window..Window>::drop.29738::hd8d0e2500a21e733
  13:     0x56111e20841f - main::h309fd0ef92de1e9cOaa
  14:     0x56111fac54e4 - sys_common::unwind::try::try_fn::h7862647019611204684
  15:     0x56111fac22f8 - __rust_try
  16:     0x56111fac5186 - rt::lang_start::h48fbdbf4c3b8b04ca4x
  17:     0x7ffab8513a3f - __libc_start_main
  18:     0x56111e1fbc78 - _start
  19:                0x0 - <unknown>
thread panicked while panicking. aborting.stack backtrace:
  Servo exited with return value -4
</pre>

**Comment by suhr** [comment]
Happens for ya.ru:

```
% ./mach run --release https://ya.ru
thread 'Constellation' panicked at 'unable to find pipeline - this is a bug', ../src/libcore/option.rs:298
stack backtrace:
   1:     0x558c3922b0f0 - sys::backtrace::tracing::imp::write::hd273f86d5325347aweu
   2:     0x558c3922e015 - panicking::default_handler::_<closure>::closure.42335
   3:     0x558c3922dabf - panicking::default_handler::h907ce0af740f5f06spy
   4:     0x558c39216af6 - sys_common::unwind::begin_unwind_inner::ha417239bcab1f2ff56s
   5:     0x558c39217258 - sys_common::unwind::begin_unwind_fmt::hac4637ed0b5bb49eb6s
   6:     0x558c3922a751 - rust_begin_unwind
   7:     0x558c392616ff - panicking::panic_fmt::he5f9c513e9167673LnK
   8:     0x558c379f6c93 - constellation::Constellation<LTF, STF>::handle_request::h15728744006046458892
   9:     0x558c379e2927 - export::compositing::constellation::Constellation<LTF, STF>::start::_<closure>::closure.22668
  10:     0x558c379e2183 - sys_common::unwind::try::try_fn::h1546569257544261773
  11:     0x558c3922a598 - __rust_try
  12:     0x558c3922686b - sys_common::unwind::try::inner_try::h62f00e3d34dbb509D3s
  13:     0x558c379e416a - boxed::F.FnBox<A>::call_box::h8839403655771340696
  14:     0x558c3922c883 - sys::thread::Thread::new::thread_start::hb3a0514b9d87669e1mx
  15:     0x7f17aa9555a5 - start_thread
  16:     0x7f17aa487adc - __clone
  17:                0x0 - <unknown>
thread 'LayoutThread PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(1) }' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError', ../src/libcore/result.rs:688
stack backtrace:
   1:     0x558c3922b0f0 - sys::backtrace::tracing::imp::write::hd273f86d5325347aweu
   2:     0x558c3922e015 - panicking::default_handler::_<closure>::closure.42335
   3:     0x558c3922dabf - panicking::default_handler::h907ce0af740f5f06spy
   4:     0x558c39216af6 - sys_common::unwind::begin_unwind_inner::ha417239bcab1f2ff56s
   5:     0x558c39217258 - sys_common::unwind::begin_unwind_fmt::hac4637ed0b5bb49eb6s
   6:     0x558c3922a751 - rust_begin_unwind
   7:     0x558c392616ff - panicking::panic_fmt::he5f9c513e9167673LnK
   8:     0x558c37c84846 - layout_thread::LayoutThread::handle_request::hbd6fd715cf0dee7bOkq
   9:     0x558c37c501ca - sys_common::unwind::try::try_fn::h11555400219251332460
  10:     0x558c3922a598 - __rust_try
  11:     0x558c3922686b - sys_common::unwind::try::inner_try::h62f00e3d34dbb509D3s
  12:     0x558c37c511aa - boxed::F.FnBox<A>::call_box::h3028057725060280698
  13:     0x558c3922c883 - sys::thread::Thread::new::thread_start::hb3a0514b9d87669e1mx
  14:     0x7f17aa9555a5 - start_thread
  15:     0x7f17aa487adc - __clone
  16:                0x0 - <unknown>
thread 'PaintThread PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(1) }' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError', ../src/libcore/result.rs:688
stack backtrace:
   1:     0x558c3922b0f0 - sys::backtrace::tracing::imp::write::hd273f86d5325347aweu
   2:     0x558c3922e015 - panicking::default_handler::_<closure>::closure.42335
   3:     0x558c3922dabf - panicking::default_handler::h907ce0af740f5f06spy
   4:     0x558c39216af6 - sys_common::unwind::begin_unwind_inner::ha417239bcab1f2ff56s
   5:     0x558c39217258 - sys_common::unwind::begin_unwind_fmt::hac4637ed0b5bb49eb6s
   6:     0x558c3922a751 - rust_begin_unwind
   7:     0x558c392616ff - panicking::panic_fmt::he5f9c513e9167673LnK
   8:     0x558c37e22139 - paint_thread::PaintThread<C>::start::h17817042636200134547
   9:     0x558c37e14411 - sys_common::unwind::try::try_fn::h3917796929493123050
  10:     0x558c3922a598 - __rust_try
  11:     0x558c3922686b - sys_common::unwind::try::inner_try::h62f00e3d34dbb509D3s
  12:     0x558c37e14f5a - boxed::F.FnBox<A>::call_box::h8465164629586486059
  13:     0x558c3922c883 - sys::thread::Thread::new::thread_start::hb3a0514b9d87669e1mx
  14:     0x7f17aa9555a5 - start_thread
  15:     0x7f17aa487adc - __clone
  16:                0x0 - <unknown>
thread 'PaintWorker' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError', ../src/libcore/result.rs:688
stack backtrace:
   1:     0x558c3922b0f0 - sys::backtrace::tracing::imp::write::hd273f86d5325347aweu
   2:     0x558c3922e015 - panicking::default_handler::_<closure>::closure.42335
   3:     0x558c3922dabf - panicking::default_handler::h907ce0af740f5f06spy
   4:     0x558c39216af6 - sys_common::unwind::begin_unwind_inner::ha417239bcab1f2ff56s
   5:     0x558c39217258 - sys_common::unwind::begin_unwind_fmt::hac4637ed0b5bb49eb6s
   6:     0x558c3922a751 - rust_begin_unwind
   7:     0x558c392616ff - panicking::panic_fmt::he5f9c513e9167673LnK
   8:     0x558c384c899a - paint_thread::WorkerThread::main::h559fa175289c5dadrwn
   9:     0x558c384c62f6 - sys_common::unwind::try::try_fn::h11899893304921864382
  10:     0x558c3922a598 - __rust_try
  11:     0x558c3922686b - sys_common::unwind::try::inner_try::h62f00e3d34dbb509D3s
  12:     0x558c384c6b96 - boxed::F.FnBox<A>::call_box::h13688735551100291871
  13:     0x558c3922c883 - sys::thread::Thread::new::thread_start::hb3a0514b9d87669e1mx
  14:     0x7f17aa9555a5 - start_thread
  15:     0x7f17aa487adc - __clone
  16:                0x0 - <unknown>
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: "SendError(..)"', ../src/libcore/result.rs:688
stack backtrace:
   1:     0x558c3922b0f0 - sys::backtrace::tracing::imp::write::hd273f86d5325347aweu
   2:     0x558c3922e015 - panicking::default_handler::_<closure>::closure.42335
   3:     0x558c3922dabf - panicking::default_handler::h907ce0af740f5f06spy
   4:     0x558c39216af6 - sys_common::unwind::begin_unwind_inner::ha417239bcab1f2ff56s
   5:     0x558c39217258 - sys_common::unwind::begin_unwind_fmt::hac4637ed0b5bb49eb6s
   6:     0x558c3922a751 - rust_begin_unwind
   7:     0x558c392616ff - panicking::panic_fmt::he5f9c513e9167673LnK
   8:     0x558c3797cc91 - compositor::IOCompositor<Window>.CompositorEventListener::handle_events::h12717050978516877207
   9:     0x558c379b367c - Browser::handle_events::h9c9ca3ecf397b098Sda
  10:     0x558c37969695 - main::h8517eb49d15e90fbNaa
  11:     0x558c3922d744 - sys_common::unwind::try::try_fn::h10561634035122728855
  12:     0x558c3922a598 - __rust_try
  13:     0x558c3922d3e6 - rt::lang_start::h632953abeabb8774Why
  14:     0x7f17aa3bd79f - __libc_start_main
  15:     0x558c3795d239 - <unknown>
  16:                0x0 - <unknown>
thread '<main>' panicked at 'You should have disposed of the pixmap properly with destroy()! This pixmap will leak!', /mnt/VCD/foo/servo/.cargo/git/checkouts/rust-layers-c02709c0157e02cd/master/src/platform/linux/surface.rs:166
stack backtrace:
   1:     0x558c3922b0f0 - sys::backtrace::tracing::imp::write::hd273f86d5325347aweu
   2:     0x558c3922e015 - panicking::default_handler::_<closure>::closure.42335
   3:     0x558c3922dabf - panicking::default_handler::h907ce0af740f5f06spy
   4:     0x558c39216af6 - sys_common::unwind::begin_unwind_inner::ha417239bcab1f2ff56s
   5:     0x558c38888c3f - sys_common::unwind::begin_unwind::h2903591670758989110
   6:     0x558c38888d40 - platform::linux::surface::PixmapNativeSurface.Drop::drop::hc1f5e1c8f8c2b325Xac
thread 'LayoutThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(0) }' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError',    7:     0x558c379786e7 - alloc..rc..Rc<layers..layers..Layer<compositing..compositor_layer..CompositorData>>::drop.29187::hd69aef1e70a6e6e4
   8:     0x558c37978835 - collections..vec..Vec<alloc..rc..Rc<layers..layers..Layer<compositing..compositor_layer..CompositorData>>>::drop.29199::h46a56cf7cb0720c9
   9:     0x558c379785c4 - alloc..rc..Rc<layers..layers..Layer<compositing..compositor_layer..CompositorData>>::drop.29187::hd69aef1e70a6e6e4
  10:     0x558c37978835 - collections..vec..Vec<alloc..rc..Rc<layers..layers..Layer<compositing..compositor_layer..CompositorData>>>::drop.29199::h46a56cf7cb0720c9
  11:     0x558c379785c4 - alloc..rc..Rc<layers..layers..Layer<compositing..compositor_layer..CompositorData>>::drop.29187::hd69aef1e70a6e6e4
  12:     0x558c3797a701 - compositing..compositor..IOCompositor<app..window..Window>::drop.29580::heb77fd6322ed2a05
  13:     0x558c379699cf - main::h8517eb49d15e90fbNaa
  14:     0x558c3922d744 - sys_common::unwind::try::try_fn::h10561634035122728855
  15:     0x558c3922a598 - __rust_try
  16:     0x558c3922d3e6 - rt::lang_start::h632953abeabb8774Why
  17:     0x7f17aa3bd79f - __libc_start_main
  18:     0x558c3795d239 - <unknown>
../src/libcore/result.rs:688
  19:                0x0 - <unknown>
thread panicked while panicking. aborting.stack backtrace:
   1:     0x558c3922b0f0 - sys::backtrace::tracing::imp::Servo exited with return value -4
```

Servo crashes on typing “test” into search field.

**Comment by asajeffrey** [comment]
Potentially related to #10017.

**Comment by asajeffrey** [comment]
Fixed by #10082.
